### PR TITLE
Update links and cleanup workflows

### PR DESCRIPTION
**Pull request by marbre**
SHARK was renamed to SHARK-Studio and SHARK-Turbine was renamed to SHARK-ModelDev. `test_models.yml` is outdated and was already deactivated.

### CODE REVIEW

File: .github/workflows/test_models.yml
Comment (ScottTodd) [inline .github/workflows/test_models.yml:1]: This entire workflow is currently disabled: https://github.com/iree-org/iree-turbine/actions/workflows/test_models.yml. We could also just delete it.

There are many things in here that are broken, like the `runs-on` which uses the old GCP cluster.

Comment (marbre) [reply .github/workflows/test_models.yml:1]: I am fine with deleting it :)

File: .github/workflows/test_shark_studio.yml
Comment (ScottTodd) [inline .github/workflows/test_shark_studio.yml:1]: Can you also update these sections?

```yml
name: Test SHARK
```

```yml
jobs:
  test-shark:
```

File: .github/workflows/test_shark_studio.yml [hunk 1]
```diff
@@ -31,10 +31,10 @@ jobs:
       - name: "Checkout SHARK"
         uses: actions/checkout@v4
         with:
-          repository: "nod-ai/SHARK"
+          repository: "nod-ai/SHARK-Studio"
           path: SHARK
```
Comment (ScottTodd) [inline .github/workflows/test_shark_studio.yml:35]: and maybe this to `SHARK-Studio`

File: .github/workflows/test_shark_studio.yml [hunk 2]
```diff
@@ -31,10 +31,10 @@ jobs:
...
 
-      - name: "Install SHARK"
+      - name: "Install SHARK Studio"
         run: |
           cd $GITHUB_WORKSPACE/SHARK
```
Comment (ScottTodd) [inline .github/workflows/test_shark_studio.yml:39]: (then update this too)

**Review by ScottTodd — APPROVED** [review]

**Comment by marbre** [comment]
Build failures are unrelated.

**Pull request merged**